File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ConfirmPasswordRule.java
package com.wesleyelliott.kubwa.rule;

/**
 * Created by wesley on 2016/07/28.
 */

public class ConfirmPasswordRule extends ConfirmRule<String> {

    public ConfirmPasswordRule() {
        super(String.class);
    }

    @Override
    public boolean isValid(String value1, String value2) {
        return value1 != null && value2 != null && value1.equals(value2);
    }
}

File: kubwa-annotations/src/main/java/com/wesleyelliott/kubwa/rule/ConfirmPasswordRule.kt
package com.wesleyelliott.kubwa.rule

/**
 * Created by wesley on 2016/07/28.
 */

class ConfirmPasswordRule : ConfirmRule<String>(String::class.java) {

    override fun isValid(value1: String?, value2: String?): Boolean {
        return value1 != null && value2 != null && value1 == value2
    }
}
